packages/docusaurus-theme-common/src/contexts/colorMode.tsx

```diff
@@ -65,10 +65,17 @@ function useContextValue(): ContextValue {
     getInitialColorMode(defaultMode),
   );
 
-  const setColorMode = useCallback((newColorMode: ColorMode) => {
-    setColorModeState(newColorMode);
-    storeColorMode(newColorMode);
-  }, []);
+  const setColorMode = useCallback(
+    (newColorMode: ColorMode) => {
+      setColorModeState(newColorMode);
+      // If switch is disabled, color mode can still be toggled through system
+      // settings. We don't store this since it's not really an active choice.
+      if (!disableSwitch) {
```

Comment:
👍 

Note we probably also want:
- to erase persisted value if site is started with `disableSwitch`
- to never persist the color when it's set from a system color mode change

See also https://github.com/facebook/docusaurus/issues/7199#issuecomment-1103929204

Reply:
> Oh, you mean, a site is deployed without disableSwitch => user visits the site and has a persisted value => site later enabled disableSwitch => we want to clear the previously stored value? Makes sense 👍

Yes, something like

```js

useEffect(() => {
  disableSwitch() && cleanStorage()
},[disableSwitch])
```

> I don't really know if that's worthwhile but I can't think of any edge case right now :P Going to do it anyways

If I select dark mode explicitly with the toggle, and put my os in light mode, then Docusaurus becomes light. I would expect it to stay dark in this case (both immediately + on refresh)